### PR TITLE
Updated the javascript build process and JS initialization

### PR DESCRIPTION
**Pull request by peeke**
The readme file and migration guide have been updated. Main changes:

- ConditionerJS has been removed in favour of VanillaJS (is easy to add back into the project, for those depending on it). This has consequences for how options are parsed (see the migration guide)
- RequireJS has been removed in favour of browserify
- Switched from babel stage-0 preset to stage-2
- Added babel es2016 preset
- Using the new bundling strategy, a separate polyfill bundle is build as useful example

Commits:
Removed require js from project
Modified js-browserify task
Added transform to require globs (e.g., whole component directory)
Removed Conditioner
Separate polyfill bundle
Removed unused dependencies, updated readme and migration guide

### CODE REVIEW

File: src/static/js/polyfill.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// From npm module
+import 'core-js/fn/dom-collections/iterator';
+import 'core-js/es6/array';
```
Comment (awolkers) [inline src/static/js/polyfill.js]: Please only import the methods that are actually used and not the complete library. 

For example use this:
```
import 'core-js/fn/array/from';
import 'core-js/fn/array/fill';
import 'core-js/fn/object/assign';
```

Comment (sanjaytwisk) [inline src/static/js/polyfill.js]: I think @awolkers has a good point here. Question remains: which polyfills will be included by default and which won't?

Comment (awolkers) [inline src/static/js/polyfill.js]: Just the bare minimum that is needed to get the bootstrap running in the supported browsers (mustard cut).

Comment (peeke) [inline src/static/js/polyfill.js]: I'll update the PR to only include the polyfills stated above somewhere later this week.

File: src/layout/default.html [hunk 2]
```diff
@@ -6,30 +6,42 @@
...
+
+            var supported =
+                'Symbol' in window &&
+                'Promise' in window &&
+                'assign' in Object &&
+                'from' in Array &&
+                'dataset' in document.createElement('div') &&
+                Symbol.iterator in NodeList.prototype;
+
+            load('/static/js/babel-helpers.js');
```
Comment (awolkers) [inline src/layout/default.html]: In IE11 and Chrome I sometimes get an issue where `main.js` is loaded before `babel-helpers.js`. It has something to do with caching. When the status 304 is return for those two files. It seems to be a better idea to make `babel-helpers` a dependency of `main.js` within this loading function.

**Pull request merged**